sdk/src/token_client.rs

```diff
@@ -0,0 +1,178 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    bcs,
+    move_types::{
+        identifier::Identifier,
+        language_storage::ModuleId,
+    },
+    rest_client::{Client as ApiClient, PendingTransaction},
+    transaction_builder::TransactionBuilder,
+    types::{
+        account_address::AccountAddress,
+        chain_id::ChainId,
+        transaction::{EntryFunction, TransactionPayload},
+        LocalAccount,
+    },
+};
+use anyhow::{Context, Result};
+use std::time::{SystemTime, UNIX_EPOCH};
+
+#[derive(Clone, Debug)]
+pub struct TokenClient<'a> {
+    api_client: &'a ApiClient,
+}
+
+impl<'a> TokenClient<'a> {
+    pub fn new(api_client: &'a ApiClient) -> Self {
+        Self { api_client }
+    }
+
+    pub async fn create_collection(
+        &self,
+        account: &mut LocalAccount,
+        name: &str,
+        description: &str,
+        uri: &str,
+        max_amount: u64,
+        options: Option<TransactionOptions>,
+    ) -> Result<PendingTransaction> {
+        let options = options.unwrap_or_default();
+
+        let chain_id = self
+            .api_client
+            .get_index()
+            .await
+            .context("Failed to get chain ID")?
+            .inner()
+            .chain_id;
+        let transaction_builder = TransactionBuilder::new(
+            TransactionPayload::EntryFunction(EntryFunction::new(
+                ModuleId::new(
+                    AccountAddress::from_hex_literal("0x3").unwrap(),
+                    Identifier::new("token").unwrap(),
+                ),
+                Identifier::new("create_collection_script").unwrap(),
+                vec![],
+                vec![
+                    bcs::to_bytes(&name).unwrap(),
+                    bcs::to_bytes(&description).unwrap(),
+                    bcs::to_bytes(&uri).unwrap(),
+                    bcs::to_bytes(&max_amount).unwrap(),
+                    bcs::to_bytes(&vec![false, false, false]).unwrap(),
+                ],
+            )),
+            SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .unwrap()
+                .as_secs()
+                + options.timeout_secs,
+            ChainId::new(chain_id),
+        )
+        .sender(account.address())
+        .sequence_number(account.sequence_number())
+        .max_gas_amount(options.max_gas_amount)
+        .gas_unit_price(options.gas_unit_price);
+        let signed_txn = account.sign_with_transaction_builder(transaction_builder);
+        Ok(self
+            .api_client
+            .submit(&signed_txn)
+            .await
+            .context("Failed to submit transfer transaction")?
+            .into_inner())
+    }
+
+    pub async fn create_token(
+        &self,
+        account: &mut LocalAccount,
+        collection_name: &str,
+        name: &str,
+        description: &str,
+        supply: u64,
+        uri: &str,
+        max_amount: u64,
+        royalty_payee_address: &AccountAddress,
```

Comment:
`AccountAddress` implements `Copy` so you can actually just pass it in without a reference if you want.